pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_alter_table failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_alter_table [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/11354782?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/11354782?buildTab=artifacts#/) on release-23.1 @ [e0e71af3fc0d357968e5dc46ca95d881bc8937e4](https://github.com/cockroachdb/cockroach/commits/e0e71af3fc0d357968e5dc46ca95d881bc8937e4):


```
=== RUN   TestTenantLogic_alter_table
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/bcf76fb9260a6174e112be124bde53cb/logTestTenantLogic_alter_table2312333507
    test_log_scope.go:79: use -show-logs to present logs inline
[05:49:43] setting distsql_workmem='98840B';
[05:49:43] setting distsql_workmem='98840B';
[05:49:45] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 18 statements
    logictestbase.go:485: statement/query skipped with reason: unsupported configuration 3node-tenant, statement/query only supports local-legacy-schema-changer (no issue given)
    logictestbase.go:485: statement/query skipped with reason: unsupported configuration 3node-tenant, statement/query only supports local-legacy-schema-changer (no issue given)
[05:49:47] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 52 statements
[05:49:50] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 61 statements
    logictestbase.go:485: statement/query skipped with reason: unsupported configuration 3node-tenant, statement/query only supports local-legacy-schema-changer (no issue given)
[05:49:52] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 75 statements
[05:49:54] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 86 statements
[05:49:57] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 93 statements
[05:49:59] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 97 statements
[05:50:01] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 104 statements
[05:50:04] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 110 statements
[05:50:06] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 113 statements
[05:50:08] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 135 statements
[05:50:10] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table: 143 statements
=== CONT  TestTenantLogic_alter_table
    logic.go:4178: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/bcf76fb9260a6174e112be124bde53cb/logTestTenantLogic_alter_table2312333507
--- FAIL: TestTenantLogic_alter_table (68.27s)
=== RUN   TestTenantLogic_alter_table/add_multiple_in_txn
    logic.go:2887: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table:2231: SELECT * FROM multipleinstmt ORDER BY id ASC;
        expected:
            1  a  a  false  NULL
            2  b  b  false  NULL
            3  c  c  false  NULL
            
        but found (query options: "") :
            4  a  a  false  NULL
            5  b  b  false  NULL
            6  c  c  false  NULL
    logic.go:2146: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table:2238: too many errors encountered, skipping the rest of the input
[05:50:38] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3901/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table with config 3node-tenant: 459 tests, 2 failures
[05:50:40] --- total progress: 459 statements
--- total: 459 tests, 2 failures
    --- FAIL: TestTenantLogic_alter_table/add_multiple_in_txn (0.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_alter_table.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30689

--- a/pkg/sql/logictest/testdata/logic_test/alter_table
+++ b/pkg/sql/logictest/testdata/logic_test/alter_table
@@ -2210,11 +2210,8 @@ COMMIT
 subtest add_multiple_in_txn
 
 statement ok
-CREATE SEQUENCE IF NOT EXISTS multipleinstmt_seq MAXVALUE 9007199254740991;
-
-statement ok
 CREATE TABLE IF NOT EXISTS multipleinstmt (
-	id INT8 DEFAULT nextval('multipleinstmt_seq') PRIMARY KEY, key STRING, value STRING
+	key STRING PRIMARY KEY, value STRING
 );
 
 statement ok
@@ -2229,23 +2226,23 @@ ALTER TABLE multipleinstmt ADD COLUMN IF NOT EXISTS a BOOL DEFAULT false;
 ALTER TABLE multipleinstmt ADD COLUMN IF NOT EXISTS b STRING;
 COMMIT;
 
-query ITTBT
-SELECT * FROM multipleinstmt ORDER BY id ASC;
+query TTBT
+SELECT * FROM multipleinstmt ORDER BY key ASC;
 ----
-1  a  a  false  NULL
-2  b  b  false  NULL
-3  c  c  false  NULL
+a  a  false  NULL
+b  b  false  NULL
+c  c  false  NULL
 
 statement ok
 ALTER TABLE multipleinstmt ADD COLUMN IF NOT EXISTS c BOOL DEFAULT true,
                            ADD COLUMN IF NOT EXISTS d STRING;
 
-query ITTBTBT
-SELECT * FROM multipleinstmt ORDER BY id ASC;
+query TTBTBT
+SELECT * FROM multipleinstmt ORDER BY key ASC;
 ----
-1  a  a  false  NULL  true  NULL
-2  b  b  false  NULL  true  NULL
-3  c  c  false  NULL  true  NULL
+a  a  false  NULL  true  NULL
+b  b  false  NULL  true  NULL
+c  c  false  NULL  true  NULL
 
 skip_on_retry
 
@@ -2412,8 +2409,8 @@ FROM (
 LEFT JOIN pg_catalog.pg_depend r ON l.table_id = r.objid;
 ----
 table_id  name               state   refobjid
-202       test_serial_b_seq  PUBLIC  201
-201       test_serial        PUBLIC  NULL
+201       test_serial_b_seq  PUBLIC  200
+200       test_serial        PUBLIC  NULL
 
 statement ok
 DROP TABLE test_serial;
@@ -2447,8 +2444,8 @@ FROM (
 LEFT JOIN pg_catalog.pg_depend r ON l.table_id = r.objid;
 ----
 table_id  name               state   refobjid
-204       test_serial_b_seq  PUBLIC  203
-203       test_serial        PUBLIC  NULL
+203       test_serial_b_seq  PUBLIC  202
+202       test_serial        PUBLIC  NULL
 
 statement ok
 ALTER TABLE test_serial DROP COLUMN b;
@@ -2463,7 +2460,7 @@ FROM (
 LEFT JOIN pg_catalog.pg_depend r ON l.table_id = r.objid;
 ----
 table_id  name         state   refobjid
-203       test_serial  PUBLIC  NULL
+202       test_serial  PUBLIC  NULL
 
 statement ok
 DROP TABLE test_serial;

